py_ecc/bls/hash.py

```diff
@@ -15,3 +18,39 @@ def hash_eth2(data: Union[bytes, bytearray]) -> Hash32:
     a future Ethereum 2.0 deployment phase.
     """
     return Hash32(hashlib.sha256(data).digest())
+
+
+def hkdf_extract(salt: Union[bytes, bytearray], ikm: Union[bytes, bytearray]) -> bytes:
+    """
+    HKDF-Expand
+
+    https://tools.ietf.org/html/rfc5869
+    """
+    return hmac.new(salt, ikm, hashlib.sha256).digest()
+
+
+def hkdf_expand(prk: Union[bytes, bytearray], info: Union[bytes, bytearray], length: int) -> bytes:
+    """
+    HKDF-Expand
+
+    https://tools.ietf.org/html/rfc5869
+    """
+    # n = cieling(length / HASH_LENGTH_BYTES)
+    n = length // HASH_LENGTH_BYTES
+    if n * HASH_LENGTH_BYTES < length:
+        n += 1
+
+    # okm = T(1) || T(2) || T(3) || ... || T(n)
+    okm = bytes(0)
+    previous = bytes(0)
+
+    for i in range(0, n):
+        # Concatenate (T(i) || info || i)
+        text = previous + info + bytes([i + 1])
+
+        # T(i + 1) = HMAC(T(i) || info || i)
+        previous = hmac.new(prk, text, hashlib.sha256).digest()
+        okm = okm + previous
```

Comment:
if you used a `bytearray` for the `okm` variable I think we could prevents some extra object copies by using `okm.extend(previous)`  Same for `previous`

Reply:
Yep I agree :)